### PR TITLE
refactor(meta): rename replace_table to replace_stream_job where appropriate

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
Although it was only for sink into table/ alter table now, actually in most places, `replace_table` is as general as `replace_stream_job`.

I noticed this after renaming `TableFragments` to `StreamJobFragments` in #19510. 

This can pave way for #18935 

Also include some minor style refactoring.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
* **#18935** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/18935?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19537** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19537?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19537?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/meta/src/barrier/command.rs [hunk 1]
```diff
@@ -84,10 +84,10 @@ pub struct Reschedule {
     pub newly_created_actors: Vec<(StreamActor, PbActorStatus)>,
 }
 
-/// Replacing an old table with a new one. All actors in the table job will be rebuilt.
-/// Used for `ALTER TABLE` ([`Command::ReplaceTable`]) and sink into table ([`Command::CreateStreamingJob`]).
+/// Replacing an old job with a new one. All actors in the job will be rebuilt.
+/// Used for `ALTER TABLE` ([`Command::ReplaceStreamJob`]) and sink into table ([`Command::CreateStreamingJob`]).
 #[derive(Debug, Clone)]
-pub struct ReplaceTablePlan {
+pub struct ReplaceStreamJobPlan {
```
Comment (xxchan) [inline src/meta/src/barrier/command.rs:90]: We can see no fields here is strongly related with `Table`.

Comment (xxchan) [reply src/meta/src/barrier/command.rs:90]: On the contrary, the proto contains `catalog.Table`

https://github.com/risingwavelabs/risingwave/blob/11-22-refactor_rename_replace_table_to_replace_stream_job/proto/ddl_service.proto#L356-L369

**Review by yezizp2012 — APPROVED** [review]

File: src/meta/src/controller/streaming_job.rs [hunk 2]
```diff
@@ -1014,57 +1000,65 @@ impl CatalogController {
...
     ) -> MetaResult<(Vec<Relation>, Vec<PbFragmentWorkerSlotMapping>)> {
-        // Question: The source catalog should be remain unchanged?
-        let StreamingJob::Table(_, table, ..) = streaming_job else {
-            unreachable!("unexpected job: {streaming_job:?}")
-        };
+        let original_job_id = streaming_job.id() as ObjectId;
 
-        let job_id = table.id as ObjectId;
+        match streaming_job {
+            StreamingJob::Table(_source, table, _table_job_type) => {
```
Comment (BugenZhao) [inline src/meta/src/controller/streaming_job.rs:1020]: Shall we `let .. else { .. }` to avoid indent change?

**Review by BugenZhao — APPROVED** [review]
ok

File: src/meta/src/controller/streaming_job.rs (hunk 2 above)
Comment (xxchan) [reply src/meta/src/controller/streaming_job.rs:1020]: will add other branch soon...

**Pull request merged**